### DISCUSSION TITLE
How to handle feature-only datasets?

### DISCUSSION THREAD

**Discussion by t13m** [lhotse-speech/lhotse#1224] [Q&A] [answered]
Wondering is there any way to handle datasets where no audio but their features are available? For example, AudioSet provides audio features extracted from youtube videos.

I'm new to lhotse. If I understand the document correctly, a typical lhotse workflow starts from constructing the recording manifest, and audio source seems required there. Am I missing something?

**Comment by desh2608** [comment, marked as answer]
Yes, a `Cut` in Lhotse can be defined using any combination of `Recording`, `Feature`, or `Supervision`. See the `CutSet.from_manifests()` method here: https://github.com/lhotse-speech/lhotse/blob/89ca0e695e88eee775286744a418d4f726ee2679/lhotse/cut/set.py#L310.

Once you have created the CutSet using the features, you can use them in the same way as you would use any regular cut set.

**Reply by desh2608** [reply]
The only thing you would need to do is to store the features as a `FeatureSet`. I would suggest looking at this class: https://github.com/lhotse-speech/lhotse/blob/89ca0e695e88eee775286744a418d4f726ee2679/lhotse/features/io.py#L258

**Reply by t13m** [reply]
OK, that's exactly what I want to know. Thank you! I'll look into it.
